source/schema.d.ts

```diff
@@ -32,12 +33,34 @@ const userMaskSettings: UserMask = {
 	created: 'show',
 	active: 'show',
 	passwordHash: 'hide',
+	attributes: ['mask', 'show']
+}
+```
+
+By default, this affects elements in array and tuple types. You can change this by passing `{recurseIntoArrays: false}` as the third type argument:
+- If `recurseIntoArrays` is set to `true` (default), array elements will be recursively processed as well.
+- If `recurseIntoArrays` is set to `false`, arrays will not be recursively processed, and the entire array will be replaced with the given value type.
+
+@example
+```
+type UserMask = Schema<User, 'mask' | 'hide' | 'show', {recurseIntoArrays: false}>;
+
+const userMaskSettings: UserMask = {
+	id: 'show',
+	name: {
+		firstname: 'show',
+		lastname: 'mask',
+	},
+	created: 'show',
+	active: 'show',
+	passwordHash: 'hide',
+	attributes: 'hide'
 }
 ```
 
 @category Object
 */
-export type Schema<ObjectType, ValueType> = ObjectType extends string
+export type Schema<ObjectType, ValueType, Options extends SchemaOptions = {recurseIntoArrays: true}> = ObjectType extends string
```

Comment:
It should not have `= {recurseIntoArrays: true}` because it will only work correctly as long as there is a single option. Which could cause bugs in the future if we add another option.